proxywasm/internal/abi_callback_configuration_test.go

```diff
@@ -65,8 +66,15 @@ func Test_proxyOnVMStart(t *testing.T) {
 	// Check ABI version. There is no return value so just make sure it doesn't panic.
 	proxyABIVersion()
 
-	// Allocate memory
-	require.NotNil(t, proxyOnMemoryAllocate(100))
+	// Allocate memory. The returned buffer has the same size as passed in
+	bufPtr := proxyOnMemoryAllocate(100)
+	buf := unsafe.Slice(bufPtr, 101)
+	for _, b := range buf[:100] {
+		require.Equal(t, byte(0), b)
+	}
+	require.Panics(t, func() {
+		_ = buf[101]
+	})
```

Comment:
can we move this out of `Test_proxyOnVMStart` into a separate `Test_proxyOnMemoryAllocate`?

Reply:
What do you think about renaming `Test_proxyOnVMStart` to `Test_pluginInitialization`? I noticed there are several ABI tested for initialization in this function so adding the other ones that Envoy would call felt natural. Or I can separate out no worries, just want to doublecheck

Reply:
works for me!

Reply:
Thanks, renamed